File: Creational/Prototype/JavaScript/shape.js
// shape.js
class Shape {
  constructor(color) {
    this.color = color;
  }

  setColor(newColor) {
    this.color = newColor;
  }

  getColor() {
    return this.color;
  }

  clone() {
    return Object.assign(Object.create(Object.getPrototypeOf(this)), this);
  }

  draw() {
    throw new Error("Subclass must implement abstract method");
  }
}

module.exports = Shape;

File: Creational/Prototype/Python/shape.py
# shape.py
import copy

class Shape:
  def __init__(self, color):
    self.color = color

  def set_color(self, new_color):
    self.color = new_color

  def get_color(self):
    return self.color

  def clone(self):
    return copy.deepcopy(self)

  def draw(self):
    raise NotImplementedError("Subclass must implement abstract method")
